tests/Plaid.Tests/Tests/PlaidClientTest.cs

```diff
@@ -24,21 +25,19 @@ public PlaidClientTest()
 				.AddEnvironmentVariables("PLAID_CONFIG_")
 				.AddJsonFile("secrets.json", optional: true)
 				.Build();
+			PlaidOptions plaidOptions = configuration.GetSection(PlaidOptions.SectionKey).Get<PlaidOptions>();
```

Comment:
Update line 25 to remove the `"PLAID_CONFIG_"`. You can test by removing `secrets.json` and setting `PLAID_ENVIRONMENT`, etc. environment variable. If this works (as it should), then update the exception messages below accordingly (`PLAID_CONFIG_ENVIRONMENT` => `PLAID_ENVIRONMENT`, etc.)

Reply:
Both are needed since the environment vars are not grouped into a "Plaid" json group for IOptions  - but I will add some code below that lets the environment vars override.

Reply:
Environment variables use _ to separate sections. So you can use `SectionKey` here, just need to remove the `_CONFIG` from the environment variable we are expecting. So it would now the `PLAID_ENVIRONMENT` instead.

Reply:
PS: If this works, update appveyor.yml to remove the `_config` in the environment variable section

Reply:
I think I can just do this (and then fix up the error checking below it):

```csharp
PlaidOptions plaidOptions = configuration.GetSection(PlaidOptions.SectionKey).Get<PlaidOptions>()
	?? new PlaidOptions()
	{
		ClientId = configuration["CLIENT_ID"],
		Secret = configuration["SECRET"],
		DefaultAccessToken = configuration["ACCESS_TOKEN"]
	};
```

The Environment check is a little more involved...
```csharp
// Since the default value of the PlaidOptions.Environment property is Sandbox, only allow the PLAID_CONFIG_ENVIRONMENT 
// to override a Sandbox value. Otherwise, the non-default value was read from secrets.json so we do not need to override 
// (as non of the other environment vars are used either)
if (!string.IsNullOrWhiteSpace(configuration["ENVIRONMENT"]) && plaidOptions.Environment == Environment.Sandbox)
{
	if (!Enum.TryParse<Environment>(configuration["ENVIRONMENT"], true, out Environment env))
	{
		throw new InvalidOperationException($"Environment configuration via PLAID_CONFIG_ENVIRONMENT is not valid. " +
			$"Actual: {configuration["ENVIRONMENT"]} ... " +
			$"Valid: {Enum.GetNames(typeof(Environment)).Aggregate((x,y) => $"{x}, {y}")}");
	}
	else
	{
		plaidOptions.Environment = env;
	}
}
```

Reply:
I'd prefer not to do that. You can leave it for now, I'll do final test using environment variables before pulling in the PR.

Reply:
ha. ok well I just committed that but I can change back. If you read in the Environment vars as a config section you'll need to make the names match the PlaidOptions Properties. If you do that, then they hierarchy of applying the env and secrets.json to the IConfiguration should "just work" ... I think. I've done that with multiple json files to allow overrides but not with env vars before.

Reply:
Agreed. Should just work. It did work the other way, so I know the concept of overlaying secrets.json and environment variables works in principle.  :)

Reply:
It works when the keys are at the root level of the JSON file as they can map directly to the suffix of the env var name. But when the JSON has nested values, you might have to have the env values be like "Plaid:ClientId" in order to map to the nested ClientId value... but I've never done it so not sure :)

Reply:
Yes, the `:` turns into `_` when using environment variables. So `PLAID_ENVIRONMENT` matches `Plaid:Environment` in the json.

Reply:
Ah... Ok, well I can switch it back tomorrow to remove the more complex Environment check I have in there and rely on that.

All three crazy config things - I feel like I learn a new way to do things on every project :)

Reply:
ok, code updated - it will fail to build until the env vars are updated to be PLAID_<name> (i.e. remove the _CONFIG_ part)